WordPressAuthenticator/Services/SocialService.swift

```diff
@@ -9,9 +9,16 @@ public enum SocialService {
         public let fullName: String
     }
 
+    public var user: User {
+        switch self {
+        case .google(let user): return user
+        case .apple(let user): return user
+        }
+    }
+
     /// Google's Signup Linked Account
     ///
-    case google(user: GIDGoogleUser)
+    case google(user: User)
```

Comment:
`apple` and `google` both have a `User` associated type now.

This kind of makes it backward to have an enum with associated type, we should have a type with an enum property

```swift
struct SocialUser {

  let fullName: String
  let email: String
  let source: SocialUser.Source // .apple, .google, ...
}
```

I've come across other bits of information in some places that could go into this type, such as the token.

I'm skeptical of working on "such a big" rewrite, though. We should be making changes to the auth layer here carefully. Also, I'd like to first remove all the Google-SDK cruft first. The downside is that we'll have to go through at least two major versions. I think it's worth it, though.